src/tcache.c

```diff
@@ -952,7 +946,8 @@ tcache_enabled_set(tsd_t *tsd, bool enabled) {
 	bool was_enabled = tsd_tcache_enabled_get(tsd);
 
 	if (!was_enabled && enabled) {
-		tsd_tcache_data_init(tsd, NULL);
+		tsd_tcache_data_init_with_bin_settings(tsd, NULL,
```

Comment:
We can rename this to `tsd_tcache_data_init` now, since bin settings are required always.